lib/AppInfo/Application.php

```diff
@@ -100,16 +101,28 @@ private function registerRedirect(IRequest $request, IURLGenerator $urlGenerator
 		}
 	}
 
-	private function registerLogin(IRequest $request, IL10N $l10n, IURLGenerator $urlGenerator, ProviderMapper $providerMapper): void {
+	private function registerLogin(
+		IRequest $request, IL10N $l10n, IURLGenerator $urlGenerator, IConfig $config, ProviderMapper $providerMapper,
+	): void {
 		$redirectUrl = $request->getParam('redirect_url');
 		$absoluteRedirectUrl = !empty($redirectUrl) ? $urlGenerator->getAbsoluteURL($redirectUrl) : $redirectUrl;
 		$providers = $this->getCachedProviders($providerMapper);
-		foreach ($providers as $provider) {
-			// FIXME: Move to IAlternativeLogin but requires boot due to db connection
-			OC_App::registerLogIn([
-				'name' => $l10n->t('Login with %1s', [$provider->getIdentifier()]),
-				'href' => $urlGenerator->linkToRoute(self::APP_ID . '.login.login', ['providerId' => $provider->getId(), 'redirectUrl' => $absoluteRedirectUrl]),
-			]);
+		$customLoginLabel = $config->getSystemValue('user_oidc', [])['login_label'] ?? '';
+		if ($customLoginLabel) {
+			foreach ($providers as $provider) {
+				OC_App::registerLogIn([
+					'name' => preg_replace('/{name}/', $provider->getIdentifier(), $customLoginLabel),
+					'href' => $urlGenerator->linkToRoute(self::APP_ID . '.login.login', ['providerId' => $provider->getId(), 'redirectUrl' => $absoluteRedirectUrl]),
+				]);
+			}
+		} else {
+			foreach ($providers as $provider) {
+				// FIXME: Move to IAlternativeLogin but requires boot due to db connection
+				OC_App::registerLogIn([
+					'name' => $l10n->t('Login with %1s', [$provider->getIdentifier()]),
+					'href' => $urlGenerator->linkToRoute(self::APP_ID . '.login.login', ['providerId' => $provider->getId(), 'redirectUrl' => $absoluteRedirectUrl]),
+				]);
+			}
 		}
```

Comment:
Would prefer to keep it shorter

```suggestion
		foreach ($providers as $provider) {
			$name = $customLoginLabel ? preg_replace('/{name}/', $provider->getIdentifier(), $customLoginLabel) : $l10n->t('Login with %1s', [$provider->getIdentifier()]);
			OC_App::registerLogIn([
				'name' => $name,
				'href' => $urlGenerator->linkToRoute(self::APP_ID . '.login.login', ['providerId' => $provider->getId(), 'redirectUrl' => $absoluteRedirectUrl]),
			]);
		}
```

Reply:
Done, i was trying to save a few condition evaluations, that should be negligible indeed.